src/radiation/radiation.cpp

```diff
@@ -463,6 +460,38 @@ void ablate::radiation::Radiation::ParticleStep(ablate::domain::SubDomain& subDo
                     utilities::PetscUtilities::checkError;  //!< Get the fields from the radsolve swarm so the new point can be written to them
                 DMSwarmRestoreField(radReturn, DMSwarmField_rank, nullptr, nullptr, (void**)&returnRank) >> utilities::PetscUtilities::checkError;
             }
+        }
+    }
+    DMSwarmRestoreField(radSearch, IdentifierField, nullptr, nullptr, (void**)&identifiers) >> utilities::PetscUtilities::checkError;
+    DMSwarmRestoreField(radSearch, DMSwarmPICField_cellid, nullptr, nullptr, (void**)&index) >> utilities::PetscUtilities::checkError;
+}
+
+void ablate::radiation::Radiation::ParticleStep(ablate::domain::SubDomain& subDomain, DM faceDM, const PetscScalar* faceGeomArray, DM radReturn) { /** Check that the particle is in a valid region */
+    PetscInt npoints = 0;
+    PetscInt nglobalpoints = 0;
+    DMSwarmGetLocalSize(radSearch, &npoints) >> utilities::PetscUtilities::checkError;
+    DMSwarmGetSize(radSearch, &nglobalpoints) >> utilities::PetscUtilities::checkError;
+
+    PetscFVFaceGeom* faceGeom;
+
+    PetscMPIInt rank = 0;
+    MPI_Comm_rank(subDomain.GetComm(), &rank);
+
+    /** Declare some information associated with the field declarations */
+    PetscInt* index;
+    struct Virtualcoord* virtualcoords;  //!< Pointer to the primary (virtual) coordinate field information
+    struct Identifier* identifiers;      //!< Pointer to the ray identifier information
+
+    /** Get all of the ray information from the particle
+     * Get the ntheta and nphi from the particle that is currently being looked at. This will be used to identify its ray and calculate its direction. */
+    DMSwarmGetField(radSearch, IdentifierField, nullptr, nullptr, (void**)&identifiers) >> utilities::PetscUtilities::checkError;
+    DMSwarmGetField(radSearch, VirtualCoordField, nullptr, nullptr, (void**)&virtualcoords) >> utilities::PetscUtilities::checkError;
+    DMSwarmGetField(radSearch, DMSwarmPICField_cellid, nullptr, nullptr, (void**)&index) >> utilities::PetscUtilities::checkError;
+
+    for (PetscInt ipart = 0; ipart < npoints; ipart++) {
```

Comment:
This just removes the segment creation and assigns the path length / cell index.